src/osd/modules/lib/osdlib.h

```diff
@@ -48,14 +53,53 @@ void osd_process_kill(void);
 
 int osd_setenv(const char *name, const char *value, int overwrite);
 
+
 /*-----------------------------------------------------------------------------
     osd_get_clipboard_text: retrieves text from the clipboard
 
     Return value:
 
         the returned string needs to be osd_free()-ed!
-
 -----------------------------------------------------------------------------*/
+
 char *osd_get_clipboard_text(void);
 
+
+/*-----------------------------------------------------------------------------
+    osd_dynamic_bind: load functions from optional shared libraries
+
+    Notes:
+
+        - Supports Mac OS X, Unix and Windows (both desktop and Windows
+          Store universal applications)
+        - A symbol can be searched in a list of libraries (e.g. more
+          revisions of a same library)
+-----------------------------------------------------------------------------*/
+
+class osd_dynamic_bind_base
+{
+public:
+	osd_dynamic_bind_base(const char *symbol, const std::vector<std::wstring> libraries);
```

Comment:
Please change this API to use UTF8 char strings, rather than wstring as per MAME convention and convert as necessary.  Also, think about efficiency: passing vector of strings by value causes lots of copying, consider using const reference.